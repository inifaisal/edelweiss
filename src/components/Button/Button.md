Default Button:

```js
<Button text='button' />
```

Primary Button:

```js
<Button variant='primary' text='button'  />
<Button variant='primary' text='button' rounded />
```

Secondary Button:

```js
<Button variant='secondary' text='button' />
```

Warning Button:

```js
<Button state='warning' text='warning button' />
```

Alert Button:

```js
<Button state='alert' text='alert button' />
```
