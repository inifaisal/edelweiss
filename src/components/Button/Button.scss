@import "../../styles/_variables";


.btn {
  cursor: pointer;
  padding: 10px;
  outline: none;
  border: none;
  color: $common-text-color;
  border-radius: 3px;

  &:hover {
    background: darken($color: $default-color, $amount: 10);
  }
  &:active {
    box-shadow: inset 1px 1px 3px 1px darken($color: $default-color, $amount: 18);
  }

  &.primary {
    background: $primary-color;
    color: #fff;
    &:hover {
      background: darken($color: $primary-color, $amount: 10);
    }
    &:active {
      box-shadow: inset 1px 1px 3px 1px darken($color: $primary-color, $amount: 18);
    }
  }

  &.secondary {
    background: $secondary-color;
    color: #fff;
    &:hover {
      background: darken($color: $secondary-color, $amount: 10);
    }
    &:active {
      box-shadow: inset 1px 1px 3px 1px darken($color: $secondary-color, $amount: 18);
    }
  }

  &.st-warning {
    color: $common-text-color;
    background: $warning-color;
    &:hover {
      background: darken($color: $warning-color, $amount: 10);
    }
    &:active {
      box-shadow: inset 1px 1px 3px 1px darken($color: $warning-color, $amount: 18);
    }
  }

  &.st-alert {
    color: #fff;
    background: $alert-color;
    &:hover {
      background: darken($color: $alert-color, $amount: 5);
    }
    &:active {
      box-shadow: inset 1px 1px 2px 1px darken($color: $alert-color, $amount: 18);
    }
  }
}
