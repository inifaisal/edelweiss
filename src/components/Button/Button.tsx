import React, { Component } from 'react';

import classNames from 'classnames';
//import './Button.scss';

export interface IButtonProps {
  text?: string;
  variant: 'primary' | 'secondary' | 'default';
  rounded?: boolean;
  state?: 'warning' | 'alert';
}

class Button extends Component<IButtonProps> {
  static defaultProps: IButtonProps = {
    variant: 'default',
  };

  render() {
    const { variant, state, rounded, ...otherProps } = this.props;
    const btnCls: string = classNames(
      'btn',
      variant,
      state && `st-${state}`,
      rounded && 'rounded'
    );

    return (
      <button className={btnCls} type='button' {...otherProps}>
        {this.props.text}
      </button>
    );
  }
}

export default Button;
