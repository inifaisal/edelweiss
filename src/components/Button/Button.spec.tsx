import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

it('renders without crashing', () => {
  const actual = shallow(
    <Button variant='primary' text='test button' rounded />
  );

  expect(actual).toMatchSnapshot();
});
