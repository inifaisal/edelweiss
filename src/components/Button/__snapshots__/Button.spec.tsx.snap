// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    rounded={true}
    text="test button"
    variant="primary"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "test button",
      "className": "btn primary rounded",
      "text": "test button",
      "type": "button",
    },
    "ref": null,
    "rendered": "test button",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test button",
        "className": "btn primary rounded",
        "text": "test button",
        "type": "button",
      },
      "ref": null,
      "rendered": "test button",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
